pkg/grpc/exec/kthread_init.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package exec
+
+import (
+	"log"
+
+	"github.com/cilium/tetragon/api/v1/tetragon"
+	"github.com/cilium/tetragon/pkg/api/processapi"
+	"github.com/cilium/tetragon/pkg/process"
+	"github.com/cilium/tetragon/pkg/reader/notify"
+)
+
+// This is used only when we add kernel threads during /proc scanning.
+// We use the same interface with all the other events to make the handling
```

Comment:
But we also add kernel threads from the clone right? your PR does that?  I'm just thinking on how to improve that comment so it reads we handle /proc scanning here and clone part of the usuall clone.

Reply:
> But we also add kernel threads from the clone right? your PR does that?

Yes, the next commit reverts a commit that didn't do that. 

This is only used for kernel threads during proc scanning. This should not be used during normal operation. The reason that I added that is that this is somewhere in the middle compared to what the exec and clone handler do. 

In that case, we don't need to generate events compared to exec. In comparison to clones, we do not have a way to provide a process name as we always inherit that from the parent.  

But this is also much more simple. There is no need to do anything related to the eventcache. I preferred to create a new (simple) handler compared to changing the existing handlers (and making them more complex).